mathenger-api/src/main/java/com/example/mathengerapi/integrations/dspace/DSpaceMessageHandler.java

```diff
@@ -31,24 +34,54 @@ public void onNewMessage(MessageSent event) {
+        var message = event.getText();
+        UUID communityId = CommandUtils.extractId(message);
+        botCommandsHandler.handleAllCollectionsOfCommunity(event.getChatId(), communityId);
+    }
+
+    private void handleAllItemsOfCollection(MessageSent event) {
+        var message = event.getText();
+        UUID collectionId = CommandUtils.extractId(message);
+        botCommandsHandler.handleAllItemsOfCollection(event.getChatId(), collectionId);
+    }
+
+    private void handlePublication(MessageSent event) {
         var message = event.getText();
-        if ("/community".equals(message)) {
-            botCommandsHandler.handleAllCommunities(event.getChatId());
-        } else if (message.matches("/community_" + UUID_REGEX)) {
-            UUID communityId = CommandUtils.extractId(message);
-            botCommandsHandler.handleAllCollectionsOfCommunity(event.getChatId(), communityId);
-        } else if (message.matches("/colpublications_" + UUID_REGEX)) {
-            UUID collectionId = CommandUtils.extractId(message);
-            botCommandsHandler.handleAllItemsOfCollection(event.getChatId(), collectionId);
-        } else if (message.matches("/publication_" + UUID_REGEX)) {
-            UUID workId = CommandUtils.extractId(message);
-            botCommandsHandler.handlePublication(event.getChatId(), workId);
-        } else if ("/help".equals(message)) {
-            botCommandsHandler.handleAllCommands(event.getChatId());
-        } else if (chatStatusService.isPrivateChat(event.getChatId())) {
+        UUID publicationId = CommandUtils.extractId(message);
+        botCommandsHandler.handlePublication(event.getChatId(), publicationId);
+    }
+
+    private void handleAllCommands(MessageSent event) {
+        botCommandsHandler.handleAllCommands(event.getChatId());
+    }
+
+    private void handleInvalidCommand(MessageSent event) {
+        var message = event.getText();
+        if (chatStatusService.isPrivateChat(event.getChatId())) {
             botCommandsHandler.handleInvalidCommand(event.getChatId(), message);
         }
     }
+
+    private void handleBotMessage(MessageSent inputEvent) {
+        List<Pair<String, Consumer<MessageSent>>> commandToConsumerList = List.of(
+                Pair.of("/community", event -> handleAllCommunities(event)),
+                Pair.of("/community_" + UUID_REGEX, event -> handleAllCollectionsOfCommunity(event)),
+                Pair.of("/colpublications_" + UUID_REGEX, event -> handleAllItemsOfCollection(event)),
+                Pair.of("/publication_" + UUID_REGEX, event -> handlePublication(event)),
+                Pair.of("/help", event -> handleAllCommands(event))
+        );
+        var message = inputEvent.getText();
+        commandToConsumerList.forEach(entry -> {
+            if (message.matches(entry.getFirst())) {
+                entry.getSecond().accept(inputEvent);
+                return;
+            }
+        });
+        handleInvalidCommand(inputEvent);
```

Comment:
Try to rewrite this part using Stream API. Something like this:
commandToConsumerList.stream()
.filter(entry -> message.matches(entry.getFirst()))
.map(Entry::getSecond)
.findFirst()
.ifPresentOrElse(consumer-> consumer.accept(inputEvent), () -> handleInvalidCommand(inputEvent))